### PR TITLE
feat: [CDS-41030]: ECS NG All Backend Changes

### PR DESCRIPTION
**Pull request by sainath-b**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35799)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 178 Issues 
* Vulnerability 24 Vulnerabilities 
 * Code issue 145 Code issues 
* Anti-pattern 9 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35799)

**Comment by alluvamsi-harness** [comment]
trigger feature-build

**Comment by alluvamsi-harness** [comment]
trigger feature-build

**Comment by PragyeshMishra01** [comment]
trigger feature-build

**Comment by PragyeshMishra01** [comment]
trigger compile

**Comment by sainath-b** [comment]
trigger ti1

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/ecs/EcsArtifactType.java [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package io.harness.delegate.task.ecs;
```
Comment (sainath-b) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/ecs/EcsArtifactType.java:1]: Is this required?

**Review by PragyeshMishra01 — APPROVED** [review]

**Review by PragyeshMishra01 — APPROVED** [review]
> Reviewed 185 of 207 files at r1, 4 of 4 files at r2, 17 of 17 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35799)* status: :shipit: complete! ✓

**Pull request merged**